p2p/peering.go

```diff
@@ -1678,3 +1724,43 @@ func (rp *RemotePeer) SendHeadersSent() bool {
 	rp.requestedHeadersMu.Unlock()
 	return sent
 }
+
+// GetInitState attempts to get initial state by sending a GetInitState message.
+func (rp *RemotePeer) GetInitState(ctx context.Context) (*wire.MsgInitState, error) {
+	const opf = "remotepeer(%v).GetInitState"
+	msg := wire.NewMsgGetInitState()
+	msg.AddTypes(wire.InitStateTSpends)
+
+	c := make(chan *wire.MsgInitState, 1)
+	newRequest := rp.addRequestedInitState(c)
+	if !newRequest {
+		op := errors.Opf(opf, rp.raddr)
+		return nil, errors.E(op, errors.Invalid, "init state is already being requested from this peer")
+	}
+
+	stalled := time.NewTimer(stallTimeout)
+	out := rp.out
+	for {
+		select {
+		case <-ctx.Done():
+			go func() {
+				<-stalled.C
+				rp.deleteRequestedInitState()
+			}()
```

Comment:
It's not clear to me why you're doing the `deleteRequestedInitState` inside a goroutine. 

If the context is done, you can just call the `deleteRequestedInitState()` directly (because no reply is expected anymore). Doing  after waiting for stalled.C to trigger is just leaking this goroutine for potentially 30 seconds.